File: iotonrails/app/models/gateway.rb

class Gateway < ApplicationRecord
  enum Status: [:Provisioning, :Active, :Suspended, :Offline, :Decommissioned]


  has_many :Site, class_name: 'Site'
  has_many :Room, class_name: 'Room'
  has_many :Devices, class_name: 'IoTDevice'
  has_many :EdgeApplications, class_name: 'EdgeApplication'
  has_many :Certificates, class_name: 'DeviceCertificate'
  has_many :DigitalTwin, class_name: 'DigitalTwin'
  has_many :NetworkProfiles, class_name: 'NetworkProfile'

end

File: iotonaspdotnet/Domain/Gateway.cs

using iotonaspdotnet.Contracts;

namespace iotonaspdotnet.Domain;

public class Gateway
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? GatewayId { get; set; }
    public virtual string? SoftwareVersion { get; set; }
    public virtual Site? Site { get; set; }
    public virtual Room? Room { get; set; }
    public virtual ICollection<IoTDevice> Devices { get; set; } = new List<IoTDevice>();
    public virtual ICollection<EdgeApplication> EdgeApplications { get; set; } = new List<EdgeApplication>();
    public virtual ICollection<DeviceCertificate> Certificates { get; set; } = new List<DeviceCertificate>();
    public virtual DigitalTwin? DigitalTwin { get; set; }
    public virtual ICollection<NetworkProfile> NetworkProfiles { get; set; } = new List<NetworkProfile>();
    public virtual DeviceStatus? Status { get; set; }

    public static Gateway FromRequest(GatewayRequest request)
    {
        return new Gateway
        {
            Id = request.Id,
            SoftwareVersion = request.SoftwareVersion,
            Status = request.Status,
        };
    }
}
